lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2204,8 +2204,55 @@ namespace Js
     {
         charcount_t count = pThis->GetLength();
 
+        const char16* inStr = pThis->GetString();
+        const char16* inStrLim = inStr + count;
+        const char16* i = inStr;
+
+        // Try to find out the chars that do not need casing (in the ASCII range)
+        if (toCase == ToUpper)
+        {
+            while (i < inStrLim)
+            {
+
+                // first range of ascii lower-case (97-122)
+                // second range of ascii lower-case (223-255)
+                // non-ascii chars (255+)
+                if ('a' <= *i)
+                {
+                    if (*i <= 'z') { break; }
+                    if ('�' <= *i) { break; }
+                }
+
+                i++;
+            }
+        } else {
+            Assert(toCase == ToLower);
+            while (i < inStrLim)
+            {
+
+                // first range of ascii uppercase (65-90)
+                // second range of ascii uppercase (192-222)
+                // non-ascii chars (255+)
+                if ('A' <= *i)
+                {
+                    if (*i <= 'Z') { break; }
+                    if ('�' <= *i)
+                    { 
+                        if (*i < '�') { break; }
+                        if ('�' < *i) { break; }
+                    }
+                }
+
+                i++;
+            }
+        }
+
+        // If no char needs casing, return immediately
+        if (i == inStrLim) return pThis;
```

Comment:
Please put the body of an if-statement in braces.

`if (i == inStrLim) { return pThis; }`